signer/tests/integration/withdrawal_reject.rs

```diff
@@ -603,3 +604,135 @@ async fn reject_withdrawal_validation_request_being_fulfilled() {
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+    let (rpc, faucet) = regtest::initialize_blockchain();
+
+    let amount = 1_000_000;
+    let signers = TestSignerSet::new(&mut rng);
+    let amounts = [
+        SweepAmounts {
+            amount,
+            max_fee: amount / 2,
+            is_deposit: true,
+        },
+        SweepAmounts {
+            amount,
+            max_fee: amount / 2,
+            is_deposit: false,
+        },
+    ];
+
+    let mut setup = TestSweepSetup2::new_setup(signers, faucet, &amounts);
+
+    let mut ctx = TestContext::builder()
+        .with_storage(db.clone())
+        .with_first_bitcoin_core_client()
+        .with_mocked_stacks_client()
+        .with_mocked_emily_client()
+        .build();
+
+    // Normal: the request has not been marked as completed in the smart
+    // contract.
+    set_withdrawal_incomplete(&mut ctx).await;
+
+    // Normal: the signer follows the bitcoin blockchain and event observer
+    // should be getting new block events from bitcoin-core. We haven't
+    // hooked up our block observer, so we need to manually update the
+    // database with new bitcoin block headers.
+    fetch_canonical_bitcoin_blockchain(&db, rpc).await;
+
+    // Different: we need to store a row in the dkg_shares table so that we
+    // have a record of the scriptPubKey that the signers control. We need
+    // this so that the donation gets picked up correctly below.
+    setup.store_dkg_shares(&db).await;
+
+    // Normal: The signers normally have a UTXO, so we add one here too. It
+    // is necessary when checking for whether the withdrawal being
+    // fulfilled by a sweep transaction that is in the mempool.
+    setup.store_donation(&db).await;
+
+    // Normal: the request and how the signers voted needs to be added to
+    // the database. Here the bitmap in the withdrawal request object
+    // corresponds to how the signers voted.
+    setup.store_withdrawal_requests(&db).await;
+    setup.store_withdrawal_decisions(&db).await;
+
+    // Normal: We do not reject a withdrawal requests until more than
+    // WITHDRAWAL_BLOCKS_EXPIRY blocks have been observed since the smart
+    // contract that created the withdrawal request has bene observed.
+    faucet.generate_blocks(WITHDRAWAL_BLOCKS_EXPIRY + 1);
+
+    // Normal: the signer follows the bitcoin blockchain and event observer
+    // should be getting new block events from bitcoin-core. We haven't
+    // hooked up our block observer, so we need to manually update the
+    // database with new bitcoin block headers.
+    fetch_canonical_bitcoin_blockchain(&db, rpc).await;
+
+    // Different: We submit a sweep transaction into the mempool so that
+    // the TestSweepSetup2 struct has the sweep_tx_info set. We also need
+    // to submit the transaction in order for
+    // `TestSweepSetup2::store_bitcoin_withdrawals_outputs` to work as
+    // expected.
+    setup.broadcast_sweep_tx(rpc);
```

Comment:
nit: it seems it's no longer the case
```suggestion
    // Different: We submit a sweep transaction into the mempool so that
    // the TestSweepSetup2 struct has the sweep_tx_info set.
    setup.broadcast_sweep_tx(rpc);
```

Reply:
I think this was updated in https://github.com/stacks-network/sbtc/pull/1411/commits/55a0aae4a01e50369815c6558035524368cc6b30.